### ISSUE TITLE
评价指标与损失的平均除数错误（按样本数 vs 按批次数）与潜在的分母为零问题

### ISSUE THREAD

**Issue by H-Tenets** [LianpoWang/DFPPNet#26]
代码在多处计算平均 MAE/RMSE 或 epoch 损失时，使用了除以数据集样本数（len(train_dataset)/len(val_dataset)/len(test_dataset)）来计算平均，但在循环中是以 batch 为单位累加每个 minibatch 的 metric（例如 mae_train.append(utils.batch_MAE(restored[0], target, False).item())），最终把这些 batch-level 值除以样本总数。这会导致数值非常小且不正确的平均（分母应当是 batch 数或累加样本数而非 dataset 长度），并且在某些情况下如果 len_valset/len_testset 计算不当或为 0 会抛异常。
为什么重要：错误的归一化导致日志/早停/模型选择依据不可信，影响训练监控、学习率调度器和 checkpoint 策略。
